source/parse.h

```diff
@@ -4161,13 +4161,8 @@ class parser
             if (term.op->type() == lexeme::LeftBracket)
             {
                 term.expr_list = expression_list(term.op);
-                if (
-                    !term.expr_list
-                    || term.expr_list->expressions.empty()
-                    )
+                if (!term.expr_list)
                 {
-                    error("subscript expression [ ] must not be empty (if you were trying to name a C-style array type, use 'std::array' instead)");
-                    next();
```

Comment:
We should still have an error here though, should we? Like the one for `(` at line 4181 below.

In fact, now that we're treating them the same, probably these two branches for `[` and `(` should be merged and do the same thing...